diff -ur jggtrans-2.2.4.orig/jggtrans.xml jggtrans-2.2.4/jggtrans.xml
--- jggtrans-2.2.4.orig/jggtrans.xml	2013-07-24 06:42:59.000000000 +0000
+++ jggtrans-2.2.4/jggtrans.xml	2013-07-24 11:08:59.571178893 +0000
@@ -154,13 +154,14 @@
  <!--
  <servers>
         <hub tls="no"/>
-        <server tls="no">217.17.45.143</server>
-        <server tls="no">217.17.45.144</server>
-        <server tls="no">217.17.45.145</server>
-        <server tls="no">217.17.45.146</server>
-        <server tls="no">217.17.45.147</server>
-        <server tls="no">217.17.41.92</server>
-        <server tls="no">217.17.41.93</server>
+        <server tls="no">91.214.237.54</server>
+        <server tls="no">91.214.237.58</server>
+        <server tls="no">91.214.237.62</server>
+        <server tls="no">91.214.237.73</server>
+        <server tls="no">91.214.237.75</server>
+        <server tls="no">91.214.237.77</server>
+        <server tls="no">91.214.237.79</server>
+        <server tls="no">91.214.237.84</server>
  </servers>
  -->
 
diff -ur jggtrans-2.2.4.orig/src/sessions.c jggtrans-2.2.4/src/sessions.c
--- jggtrans-2.2.4.orig/src/sessions.c	2013-07-24 06:42:59.000000000 +0000
+++ jggtrans-2.2.4/src/sessions.c	2013-07-24 11:12:20.761053661 +0000
@@ -209,7 +209,49 @@
 		gg_servers=g_list_append(gg_servers, server);
 
 		server=g_new(GgServer, 1);
-		inet_aton("217.17.45.143", &server->addr);
+		inet_aton("91.214.237.54", &server->addr);
+		server->port=8074;
+		server->tls=0;
+		gg_servers=g_list_append(gg_servers, server);
+
+		server=g_new(GgServer, 1);
+		inet_aton("91.214.237.58", &server->addr);
+		server->port=8074;
+		server->tls=0;
+		gg_servers=g_list_append(gg_servers, server);
+
+		server=g_new(GgServer, 1);
+		inet_aton("91.214.237.62", &server->addr);
+		server->port=8074;
+		server->tls=0;
+		gg_servers=g_list_append(gg_servers, server);
+
+		server=g_new(GgServer, 1);
+		inet_aton("91.214.237.73", &server->addr);
+		server->port=8074;
+		server->tls=0;
+		gg_servers=g_list_append(gg_servers, server);
+
+		server=g_new(GgServer, 1);
+		inet_aton("91.214.237.75", &server->addr);
+		server->port=8074;
+		server->tls=0;
+		gg_servers=g_list_append(gg_servers, server);
+
+		server=g_new(GgServer, 1);
+		inet_aton("91.214.237.77", &server->addr);
+		server->port=8074;
+		server->tls=0;
+		gg_servers=g_list_append(gg_servers, server);
+
+		server=g_new(GgServer, 1);
+		inet_aton("91.214.237.79", &server->addr);
+		server->port=8074;
+		server->tls=0;
+		gg_servers=g_list_append(gg_servers, server);
+
+		server=g_new(GgServer, 1);
+		inet_aton("91.214.237.84", &server->addr);
 		server->port=8074;
 		server->tls=0;
 		gg_servers=g_list_append(gg_servers, server);
